lib/search_logic.rb

```diff
@@ -2,6 +2,18 @@ module SearchLogic
     SEARCH_LIMIT = 20
 
     class << self
+
+        def extract_city_place_id(json_data)
+          return nil unless json_data && json_data['results'] && json_data['results'][0]
+          json_data['results'][0]['place_id']
```

Comment:
Accessing deeply nested hashes is always ugly in Ruby without a helper function, but there's a slightly shorter version of this:

```suggestion
          json_data.try(:[], 'results')&.first.try(:[], 'place_id')
```